glean-core/src/metrics/mod.rs

```diff
@@ -16,17 +19,20 @@ use crate::Glean;
 
 pub use self::boolean::BooleanMetric;
 pub use self::counter::CounterMetric;
+pub use self::datetime::DatetimeMetric;
 pub use self::string::StringMetric;
 pub use self::string_list::StringListMetric;
+pub use self::time_unit::TimeUnit;
 pub use self::uuid::UuidMetric;
 
 #[derive(Serialize, Deserialize, Debug, Clone)]
 pub enum Metric {
-    String(String),
     Boolean(bool),
     Counter(i32),
-    Uuid(String),
+    Datetime(DateTime<Utc>, TimeUnit),
```

Comment:
I think we want to record the offset from Utc here, not drop it, so use `DateTime<FixedOffset>`?